modules/stochastic_tools/src/reporters/BiFidelityActiveLearningGPDecision.C

```diff
@@ -0,0 +1,88 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BiFidelityActiveLearningGPDecision.h"
+
+registerMooseObjectAliased("StochasticToolsApp",
+                           BiFidelityActiveLearningGPDecision,
+                           "BFActiveLearningGPDecision");
+
+InputParameters
+BiFidelityActiveLearningGPDecision::validParams()
+{
+  InputParameters params = ActiveLearningGPDecision::validParams();
+  params.addClassDescription("Perform active learning decision making in bi-fidelity modeling.");
+  params.addRequiredParam<ReporterName>("outputs_lf",
+                                        "Value of the LF model output from the SubApp.");
+  return params;
+}
+
+BiFidelityActiveLearningGPDecision::BiFidelityActiveLearningGPDecision(
+    const InputParameters & parameters)
+  : ActiveLearningGPDecision(parameters),
+    _outputs_lf(getReporterValue<std::vector<Real>>("outputs_lf", REPORTER_MODE_DISTRIBUTED))
+{
+  // Create communicator that only has processors with rows
+  _communicator.split(
+      sampler().getNumberOfLocalRows() > 0 ? 1 : MPI_UNDEFINED, processor_id(), _local_comm);
+}
+
+bool
+BiFidelityActiveLearningGPDecision::facilitateDecision()
+{
+  for (dof_id_type i = 0; i < _inputs.size(); ++i)
+  {
+    _gp_mean[i] = _gp_eval.evaluate(_inputs[i], _gp_std[i]);
+    _flag_sample[i] = !learningFunction(_outputs_lf_batch[i] + _gp_mean[i], _gp_std[i]);
+  }
+
+  for (const auto & fs : _flag_sample)
+    if (!fs)
+      return false;
```

Comment:
Is there a reason why flag_sample is negated twice? 
```
_flag_sample[i] = !learningFunction(_outputs_lf_batch[i] + _gp_mean[i], _gp_std[i]);
```
and
```
if (!fs)
      return false;
```
Is it because something outside of this object needs it with the given sign?

Reply:
I just followed the logic from the base class. Looks like the even if the `!` are removed, we should get same behavior.